src/bundle/ui-dev/src/modules/common/thumbnail/thumbnail.tsx

```diff
@@ -1,9 +1,15 @@
-import React from 'react';
-import PropTypes from 'prop-types';
-
 import Icon from '../icon/icon';
 
-const Thumbnail = ({ thumbnailData, iconExtraClasses, contentTypeIconPath }) => {
+interface ThumbnailProps {
+    thumbnailData: {
+        mimeType: string;
+        resource: string;
+    };
+    iconExtraClasses?: string;
+    contentTypeIconPath?: string | null;
+}
+
+const Thumbnail = ({ thumbnailData, iconExtraClasses = '', contentTypeIconPath = null }: ThumbnailProps) => {
```

Comment:
I would consider using different notation, in which props type is easier to spot: WDYT?
```suggestion
const Thumbnail: React.FC<ThumbnailProps> = ({ thumbnailData, iconExtraClasses = '', contentTypeIconPath = null }) => {
```